gradio/components/file_explorer.py

```diff
@@ -63,10 +63,10 @@ def __init__(
     ):
         """
         Parameters:
-            glob: The glob-style pattern used to select which files to display, e.g. "*" to match all files, "*.png" to match all .png files, "**/*.txt" to match any .txt file in any subdirectory, etc. The default value matches all files and folders recursively. See the Python glob documentation at https://docs.python.org/3/library/glob.html for more information.
+            glob: The glob-style pattern(s) used to select which files to display. Can be a single string pattern (e.g. "*.png") or a list of patterns (e.g. ["*.png", "*.jpg", "*.txt"]). When a list is provided, files matching ANY of the patterns will be displayed. The default value "**/*" matches all files and folders recursively. See the Python glob documentation at https://docs.python.org/3/library/glob.html for more information.
```

Comment:
Regex would be more flexibile but would require a new parameter and we'd have to deal with users trying to use both parameters. For now, I've just expanded to include a list of globs.